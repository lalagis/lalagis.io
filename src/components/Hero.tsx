import { useState } from "react"
import Socials from "./Socials"

export default function Hero() {
  const [avatarUrl] = useState('https://github.com/lalagis.png')

  return (
    <header className="mt-12 md:mt-18 flex flex-row items-center justify-between">
      <img src={avatarUrl} className="w-72 h-72 rounded-full border-solid border border-gray-200 drop-shadow-md mr-1 flex-none sm:block hidden" />

      <div className="flex flex-col gap-y-3 mx-auto">
        <p className="font-semibold text-emerald text-4xl mb-2">嗨👋，欢迎来到Lulala的个人站</p>
        <div className="grid grid-cols-2 divide-x text-center">
          <p className="font-bold my-2">全栈&GIS开发</p>
          <p className="font-bold my-2">时空数据分析</p>
        </div>
        <p className="text-center font-light mt-2 mb-1">2021-2025 深圳大学 地理空间信息工程本</p>
        <p className="text-center font-light mb-2 mt-1">2021-2023 深圳科荔软件有限公司 开发</p>

        <Socials />
      </div>
    </header>
  )
}