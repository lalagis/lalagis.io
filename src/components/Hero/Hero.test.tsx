import { describe, expect, test } from "vitest"
import renderer from 'react-test-renderer'
import Hero from "."

describe('Hero', () => {
  test('load hero', () => {
    const component = renderer.create(<Hero />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})