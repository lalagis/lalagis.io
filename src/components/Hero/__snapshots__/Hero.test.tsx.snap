// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero > load hero 1`] = `
<header
  className="mt-12 md:mt-18 flex flex-row items-center justify-between"
>
  <img
    className="w-72 h-72 rounded-full border-solid border border-gray-200 drop-shadow-md mr-1 flex-none sm:block hidden"
    src="https://github.com/lalagis.png"
  />
  <div
    className="flex flex-col gap-y-3 mx-auto"
  >
    <p
      className="font-semibold text-emerald text-4xl mb-2"
    >
      嗨👋，欢迎来到Lulala的个人站
    </p>
    <div
      className="grid grid-cols-2 divide-x text-center"
    >
      <p
        className="font-bold my-2"
      >
        全栈&GIS开发
      </p>
      <p
        className="font-bold my-2"
      >
        时空数据分析
      </p>
    </div>
    <p
      className="text-center font-light mt-2 mb-1"
    >
      2021-2025 深圳大学 地理空间信息工程本
    </p>
    <p
      className="text-center font-light mb-2 mt-1"
    >
      2021-2023 深圳科荔软件有限公司 开发
    </p>
    <div
      className="mt-2 flex flex-row flex-wrap gap-y-1"
    >
      <a
        className="inline-flex px-3 py-2 mr-2 rounded-md bg-gray-50 transition-colors decoration-none hover:bg-gray-700 dark:hover:bg-white dark:hover:text-gray-900 hover:text-white dark:bg-gray-50/10"
        href="https://github.com/lalagis"
        target="_blank"
      >
        <div
          className="text-xl"
        >
          <div
            className="i-akar-icons:github-fill"
          />
        </div>
        <div
          className="text-sm ml-1"
        >
          GitHub
        </div>
      </a>
      <a
        className="inline-flex px-3 py-2 mr-2 rounded-md bg-gray-50 transition-colors decoration-none hover:bg-[#5D32B5] dark:hover:bg-purple dark:hover:text-gray-900 hover:text-white dark:bg-gray-50/10"
        href=""
        target="_blank"
      >
        <div
          className="text-xl"
        >
          <div
            className="i-simple-icons:obsidian"
          />
        </div>
        <div
          className="text-sm ml-1"
        >
          Docs&Blogs(暂不部署)
        </div>
      </a>
      <a
        className="inline-flex px-3 py-2 mr-2 rounded-md bg-gray-50 transition-colors decoration-none hover:bg-[#ea4c89] hover:text-white dark:bg-gray-50/10"
        href="https://dribbble.com/Cirno339"
        target="_blank"
      >
        <div
          className="text-xl"
        >
          <div
            className="i-akar-icons:dribbble-fill"
          />
        </div>
      </a>
      <a
        className="inline-flex px-3 py-2 mr-2 rounded-md bg-gray-50 transition-colors decoration-none hover:bg-[#f7c104] hover:text-white dark:bg-gray-50/10"
        href="https://www.figma.com/@cirno339"
        target="_blank"
      >
        <div
          className="text-xl"
        >
          <div
            className="i-akar-icons:figma-fill"
          />
        </div>
      </a>
      <a
        className="inline-flex px-3 py-2 mr-2 rounded-md bg-gray-50 transition-colors decoration-none hover:bg-[#fb7299] hover:text-white dark:bg-gray-50/10"
        href="https://space.bilibili.com/11636897"
        target="_blank"
      >
        <div
          className="text-xl"
        >
          <div
            className="i-simple-icons:bilibili"
          />
        </div>
      </a>
      <a
        className="inline-flex px-3 py-2 mr-2 rounded-md bg-gray-50 transition-colors decoration-none hover:bg-[#1b2838] hover:text-white dark:bg-gray-50/10"
        href="https://steamcommunity.com/id/WWWW1123"
        target="_blank"
      >
        <div
          className="text-xl"
        >
          <div
            className="i-simple-icons:steam"
          />
        </div>
      </a>
    </div>
  </div>
</header>
`;
